src/pods/BasicUserParser.ts

```diff
@@ -0,0 +1,55 @@
+import { parse } from 'querystring';
+import type { Representation } from '../ldp/representation/Representation';
+import { APPLICATION_URL_ENCODED } from '../util/ContentTypes';
+import { UnsupportedHttpError } from '../util/errors/UnsupportedHttpError';
+import { readableToString } from '../util/StreamUtil';
+import type { User } from './User';
+import { UserParser } from './UserParser';
+import Dict = NodeJS.Dict;
+
+const requiredUserKeys: (keyof User)[] = [ 'username', 'webId', 'name' ];
+const optionalUserKeys: (keyof User)[] = [ 'email' ];
+const userKeys: Set<keyof User> = new Set(requiredUserKeys.concat(optionalUserKeys));
+
+/**
+ * A parser that extracts User data from an application/x-www-form-urlencoded body.
+ */
+export class BasicUserParser extends UserParser {
```

Comment:
@joachimvh I'm thinking we should actually build for JSON(-LD) first in the future to have an API-first approach. That is: let's set up an API endpoint, and then an HTML page that will use the API via JavaScript. Also given that automated agents will register.

Form encoding is only really needed when directly POSTing from HTML.

Reply:
Sure, I didn't really know which content-type to take here so I just copied NSS. Can easily switch it to JSON.